packages/video_player/video_player_platform_interface/test/method_channel_video_player_test.dart

```diff
@@ -141,8 +141,41 @@ void main() {
         formatHint: VideoFormat.dash,
       ));
       expect(log.log.last, 'create');
+      expect(log.createMessage?.asset, null);
       expect(log.createMessage?.uri, 'someUri');
+      expect(log.createMessage?.packageName, null);
       expect(log.createMessage?.formatHint, 'dash');
+      expect(log.createMessage?.httpHeaders, null);
+      expect(textureId, 3);
+    });
+
+    test('create with network (empty headers)', () async {
+      final int? textureId = await player.create(DataSource(
+        sourceType: DataSourceType.network,
+        uri: 'someUri',
+        httpHeaders: {},
+      ));
+      expect(log.log.last, 'create');
+      expect(log.createMessage?.asset, null);
+      expect(log.createMessage?.uri, 'someUri');
+      expect(log.createMessage?.packageName, null);
+      expect(log.createMessage?.formatHint, null);
+      expect(log.createMessage?.httpHeaders, {});
```

Comment:
Note that you won't need two different test cases once it's not nullable (which is a feature, since it indicates that there aren't two paths for the same thing).